### ISSUE TITLE
What is Event Knowledge Graph: A Survey

### ISSUE THREAD

**Issue by dreamingjudith** [dreamingjudith/paper-reading-list#35]
# What is Event Knowledge Graph: A Survey

[Arxiv](https://arxiv.org/abs/2112.15280)

# History View

## Brief history

![4BB70B58-9146-43B6-A87E-D587532B4BA8](https://user-images.githubusercontent.com/20284375/194829667-c031447a-5e69-4238-8f9c-33a6b413dabf.jpeg)

## Definitions

||||
|-|-|-|
| Nodes | event |  -|
|  -| entity |  -|
| Edges | event-event relation | temporal <br />causal <br />conditional <br />thematic |
|  -| event-entity relation | represents the arguments of events <br />i.e., the edges are the argument roles of the entities to the linked events |
| - | entity-entity relation | e.g., spouse, place of birty, country |

![FF03A898-5F1E-41FD-ABA5-E9C425900F04](https://user-images.githubusercontent.com/20284375/194829701-0d78ad40-fd6d-4e93-b34d-17c5c88b7300.jpeg)

- Event trigger: a word or span that most clearly expresses the event, i.e., indicates the event type → 약간 relation 개념
- Event argument: an ***entity*** or span that plays a specific role in the event

![3B3EDA5F-40C4-4F59-B870-7CFDF56B6F83](https://user-images.githubusercontent.com/20284375/194829754-6a2d1868-919f-4755-a921-244be91a10cd.jpeg)

# 3 Ontology View

## 3.1 Event Schema Induction

It automatically extracts *event types* (e.g., Start-position, Nominate) and their *argument roles* (e.g., person, nominee, position, time) from text

| | |
| --- | --- |
| supervised | - Pattern matching <br />- Syntax-driven techniques <br />- Classifier for each argument role using semantic and discourse features |
| semi-supervised | - using SVM to identify relevant sentences for the domain of interest → extracting domain-relevent event schema via ranking by semantic affinity <br />- vector quantized variational autoencoder to learn an embedding for each seen or unseen event type → enforce the reconstruction of each event trigger conditioned on its event type distribution |
| unsupervised | - pointwise mutual information to measure the distances between events and clustered events <br />- co-occurence statistics of (s,p,o) pairs to build a graph weighted by the symmetric conditional probabilities of the involved triple pairs <br />- Generative model for schema inductino similar to LDA + entity disambiguation <br />- representation learning: graph-based model (entity embedding for argument role), word embeddings using pretrained language model (embeddings to cluster verbs) |

![67E3DF0A-404A-4CF9-B2D5-FABF2CB079C2](https://user-images.githubusercontent.com/20284375/194829789-a1b18026-a58f-43d9-87d0-e722d121e9ed.jpeg)

## 3.2 Script Induction

- script: a stereotypical structure of event schema that expresses a specific scenario
- Methods
    - using the most frequent headword in an entity coreferential chain as the argument role of the this entity
    - collecting descriptions of script-specific event sequences form volunteers to compute a graph representation of the script’s temporal structure
    - joint distribution over the words in a document and their script assignments
        - script HMM: model script transition and emit events
        - event HMM: model event transition within a script and emit argument roles
    - HMM with states are event types in scripts, and observations are sentences describing the event instances
    - Bayesian network-based method to consider causal effect between events

## 3.3 EKG Schema Induction

- path language model to build an event graph schema repository
- schema as hidden knowledge to guide the generation of event graphs and learned via maximizing the probability of these instance graphs

# 4 Instance View

## 4.1 Event Acquisition

![6A1085D8-B36A-4BB2-8A33-17220CDC21EA](https://user-images.githubusercontent.com/20284375/194829814-81c41a83-7175-4c36-b236-23a71819fef0.jpeg)

### 4.1.1 Event Extraction

- 추출 결과에 따른 분류
    - Trigger detection
    - Argument extraction
- 일반적인 구분
    - Schema-based event extraction
        - Sentence-level event extraction
            - Intra-subtasks features
                - Dynamic multi-pooling CNN
                - bidirectional dynamic multi-pooling LSTM
                - BERT with multi-classifier for trigger detection + BERT with multiple sets of binary classifier for argument extraction
                - Using event-event relation to improve trigger detection based on BERT
                - Machine reading comprehension framework upon BERT based on question answering to extract events
            - Inter-subtasks features
            → usually follow a joint framework, easing error propagation
                - bidirectinoal gated recurrent units and FCN + memory vector
                - dependency bridges in RNN + tensor layer
                - attention-based GCN with syntactic tree of the sentence
            - Inter-IE features
                - bidirectional RNN → word embedding
                FCN and softmax based classifier
                - BERT sentence encoding, span graph, FCN-based classifier
                - local score for all the candidate triggers, entities, and their pairwise links ← BERT embeddings and FCN-based classifiers
                → beam search-based decoder
        - Document-level event extraction
            - handcrafted features
            - statistical learning methods
            - Distantly supervised method + BiLSTM + CRF
            - path-expanding subtask using entity-based directed acyclic graph
            - GNN with graphs of sentences and entities
            - multi-layer bidirectional network
    - Schema-free event extraction (Open-domain event extraction)
        - dependency path between verbs and other syntactic elements
        - Finding nouns and verbs which is matching with existing schema for candidate event triggers → identifying via manually-selected semantic relations → clustering by embedding → extracting schema → mapping with existing schemas
        - Joint probability distribution of a news cluster (headwords, contextual features, latent event type) → clustering and applying a series of rules
        - extracting from social media stream (especially tweet streams)
            - extracting event keywords based on frequency → spatial density distribution over the usage ratios at particular location → clustering
            - Fusing tweets and related web pages and applying CRF
            - non-parametric Bayesian mixture model from Twitter to disambiguate mentions for specific entity
            - BiLSTM + control gate + CRF layers

### 4.1.2 Event Relation Extraction

![E19A9BC2-C9E6-439C-A47A-1728505FA5A4](https://user-images.githubusercontent.com/20284375/194829856-c79ae92c-bc23-4a2a-903d-e8ff463bc76f.jpeg)

- Event relation의 종류
    - Event temporal relations → need to consider global consistency
    - Event causal relations
- 추출 방법
        
  ![FDC1B124-3E7A-4453-BFC0-69124BB9B309](https://user-images.githubusercontent.com/20284375/194829890-26c04872-46c5-4af7-9da4-93abf1f7a405.jpeg)
 
- temporal attributes of events (tense, event type, modality, polarity, and other linguistic features) → SVM classifier
- graph representation which encodes lexical, syntactic, and semantic information → extracting graph patterns → sorting by relevance to determine the causality between events
- restricted hidden naive Bayes model ← new feature of causal connectives from the similarities of the syntactic dependency structures
- considering global consistency
    - cascade architecture ← inferring all the transitive relations from the results of earlier classifiers
    - Markov logic model + weighted first-order logic formulae
    - local classifier with feedback by performing global inference in each round of learning
    - ILP with lexical, potential causal, syntactic features from document-level
- Neural network methods
    - FCN and softmax classifier
    - SVM-based algorithm
    - ILP
    - BiLSTM along dependency paths of event sentences
    - Siamese network to a temporal commonsense knowledge base
    - knowledge-oriented CNN
    - BiLSTM + softmax distribution + SVM-based layer
    - Using differentiable constrained learning layer instead of SVM
    - Using BERT

### 4.1.3 Event Coreference Resolution

![46172F81-7BD7-4797-846D-D409C770A08F](https://user-images.githubusercontent.com/20284375/194829952-9f641ab7-5212-41d2-847c-a2730072b2c5.jpeg)

- Unsupervised
    - rule-based approaches
    - Maximum entropy, non-parametric Bayesian estimation, cosine similarity and etc.
    - unsupervised probabilistic model
    - expectation-maximization algorithm
- Semi-supervised
    - active-learning based + maximum uncertainty, maximum expected judgment error, exploration and exploitation for manual annotation
    - similarity between candidate trigger and event type + similarity between event embedding (trained from external text and then concatenated)
- Supervised
    - Event-pair models (binary classifier)
        - feature learner: CNN (from embeddings of event trigger + left/right neighbor)
        coreference scorer: FCN + logistic regression classifier
        - feature learner: 2 neural network
        coreference scorer: FCN
        - feature learner: BERT + semantic role label embedding
        coreference score: FCN + softmax
        - lexical resource + pretrained word embeddings
    - Event-ranking models
        - antecedent vector + log-linear model
        - structure graph of the events, entities, and words for each document + GCN + FCN-based coreference scorer

![D8CE1923-8000-468B-A8AB-D227FA7B1C16](https://user-images.githubusercontent.com/20284375/194829980-c4067f8e-7498-406d-afa6-5c8880e5b0dd.jpeg)

### 4.1.4 Event Argument Completion

![8239F42C-2013-4F95-B580-BD5287042ABA](https://user-images.githubusercontent.com/20284375/194830004-3f4090fe-f5a9-4ac6-a05d-80cf992a7305.jpeg)

- embeddings of [argument role-argument] pairs + target argument role → predict missing arguments
- replacing sum to weighted sum / predicting missing arguments and argument roles
- time → event type embeddings
- assign each timestamps with hyperplane → trasnlational score of arguments via event type
- maintaining the smoothness between adjacent hyperplanes by minimizing the Euclidean distances
- tensor decomposition (event type, time, other argument)
- using beginning/end time as dual complex embeddings → mapping other arguments and combine the translation-based scores
- arguments → hyperplane of its event type → projection result → weighted sum (weights from real numbers projected from its argument roles)
- FCN-likelihood of two arguments co-participate in a common event
- tensor-based compatibility between each argument role and all the involved arguments
- event tensor
- partially sharing embeddings and over-parameterization by sparsifying the tensor
- Neural network method
    - CNN for embeddings of <argument role-argument> pair
    - feature learner: GCN for event graph → embedding
    filler predictor: Transformer

## 4.2 EKG-related Representative Graphs/Systems

![C2C2A1DA-BDC6-4F3B-99AF-883FFC58CCFB](https://user-images.githubusercontent.com/20284375/194830065-13c27bb3-d360-4767-b02d-5b0633e8a74e.jpeg)

# 5 Application View

### 5.1.1 Script Event Prediction

- Definition: to predict the subsequent script events given the historical scripts
- Formalized as script coherence evaluation
- Aspects for improving this model
    - Event representation
        - using (subject, predicate, object) triples instead of (subject, predicate) and (predicate, object) pairs
        - Adding indirect object into triples
        - Adding linearly transformed predicate and argument embeddings
        - Adding embeddings of event sentences
        - Tensor-based models
        - Using intent and sentiment of the event
        - probabilistic models, FCN, …
        - Event discourse relation → relation-aware event embeddings
        - Event graph + GNN
    - Script modeling
        - PMI, Bi-gram, FCN, cosine similarity, translation-based composition, …
        - Language model with evant chain in the script
        - Neural network-based probabilistic model or LSTM
        - Aggregating the results from multiple event chains
        - Event graph + GNN
    - Evaluation
        - Narrative cloze test
        - Adversarial narrative cloze test
        - Multiple Choice Narrative Cloze (MCNC)
        - Multiple Choice Narrative Sequences (MCNS)
        - Multiple Choice Narrative Explanation (MCNE)

### 5.1.2 Temporal KG Prediction

- Definition: To predict events for future timestamps given the historical temporal KG

![570FF375-91C8-4A05-8BF8-C5BF9EBCE4D3](https://user-images.githubusercontent.com/20284375/194830100-f7c15c3c-463a-4f88-8d3a-07b6c9e0a609.jpeg)

- Graph sequence-based methods: Extracting related subgraph sequences in history modeling
    - extracting historical events of given subject and predicate, or subject
    - iteratively sample relevant edges of argments included in the subgraph
    - considering as sequential decision and search from history via reinforcement learning
    - using entire graph sequence from the last several timestamps
    - encoding continuous dynamics: neural ordinary differential equation → GCN
- Temporal point process-based methods: using conditional intensity function
    - manually specifying → extending to deep neural networks
    - RNN to learn the evolutionary embeddings of argument + multivariate point process
    - aggregating object embeddings in the concurrent events related to the given subject → hidden vector + time as a random variable
    Hawkes process + continuous-time LSTM

### 5.1.3 Other Basic Applications

- Timeline generation
- Abductive reasoning

## 5.2 Deep Applications

- Search
- Question answering
- Recommendation
- Financial quantitative investment
- Text generation
